Userland/Libraries/LibGUI/IconView.cpp

```diff
@@ -246,9 +246,9 @@ void IconView::mouseup_event(MouseEvent& event)
     AbstractView::mouseup_event(event);
 }
 
-bool IconView::update_rubber_banding(const Gfx::IntPoint& position)
+bool IconView::update_rubber_banding(const Gfx::IntPoint& input_position)
 {
-    auto adjusted_position = to_content_position(position);
+    auto adjusted_position = to_content_position(widget_inner_rect().clamp(input_position));
```

Comment:
This API seems weird, or maybe I just can't wrap my head around it - isn't it more like *clamping a point to a rect*, rather than *clamping a rect to a point*, i.e.:

```suggestion
    auto adjusted_position = to_content_position(input_position.clamp(widget_inner_rect()));
```

?

Reply:
Hmm, i don't know any OO clamping API, they're all just straight functions.
The rationale is more like the rect is a cookie cutter/tool i'm using to clamp the point.

Reply:
And if you put it as a sentence i think my way makes more sense:
`rect.clamp(point)` → rect clamps point
vs
`point.clamp(rect)` → point is clamped by rect

But again, that could go either way.